### ISSUE TITLE
Pull to refresh

### ISSUE THREAD

**Issue by sashei** [mozilla-lockbox/lockbox-ios#330]
from both the empty state and the entries list

1. pull down partially does not sync but provides a visual indicator
2. pull down fully (a "full" pull) does then initiate sync
3. in both cases, there is a 'snap back' interaction expected here
4. after "full" pull, the spinning/loading indicator stays in place until sync is complete
5. after sync is complete, the indicator/table collapses back to just search bar + list
6. after sync is complete, newly sync'd items appear at top of table with visual highlight (nice to have)

Related: #367 (background color)

**Comment by joeyg** [comment]
@sashei Is the idea that this would initiate a DataStore sync and the spinner would go away when done syncing? If the app syncs when it launches should it start with the pull to refresh spinner visible?

**Comment by joeyg** [comment]
not sure about the interaction between this and #346.

**Comment by changecourse** [comment]
@joeyg the spinner modal will be the same for both instances; 

#346 will support what happens when the sync call is made for the first time, and future instances (if needed) where sync is called upon subsequent logins. 

This story will support the user-driven ability to refresh, via pulling down on the list, which will invoke a sync request and corresponding loader until sync is complete... I will provide designs in the description to support this action.

**Comment by changecourse** [comment]
Actually, I may double back a bit on what I previously said... I think our pull to refresh should function slightly differently than our initial sync modal... 

Given that the updates via a pull to refresh should be lightweight, perhaps we handle it in a manner that doesn't take over the screen, and serves as more of a background sync call, where we allow it to be interruptible, as opposed to the more 'required' nature of an initial sync? See attached designs. 

<img width="1879" alt="screen shot 2018-05-14 at 4 43 54 pm" src="https://user-images.githubusercontent.com/6164801/40027244-571ee8a6-5796-11e8-84cc-7a9d49822d80.png">

Thoughts from the current participants on this issue? :)

**Comment by devinreams** [comment]
@changecourse I think there's a few things going on, can you help detail this kind of stuff so we don't assume/overlook anything? Here's a quick pass:

1. pull down partially does not sync but provides a visual indicator
2. pull down fully (a "full" pull) does then initiate sync
3. in both cases, there is a 'snap back' interaction expected here
4. after "full" pull, the spinning/loading indicator stays in place until sync is complete
5. after sync is complete, the indicator/table collapses back to just search bar + list
6. after sync is complete, newly sync'd items appear at top of table with visual highlight

@changecourse two things:

1. is that list accurate and complete?
2. I don't know about the last requirement

@sashei do you know if we know what is newly sync'd and added to the table? is that straight forward enough and feasible (easy) to include as part of this story?

**Comment by changecourse** [comment]
@devinreams yes, this looks to be representative of everything... 

An indication of the updated/new items would be a nice affordance (post-sync update) and I could certainly come up with something to address that, though that could also be a new issue/enhancement down the line too.

**Comment by devinreams** [comment]
>this looks to be representative of everything...

@changecourse OK. Added to the top of the issue description and noted accordingly. I also added a reference to #367.

>>An indication of the updated/new items would be a nice affordance

OK, then the feature is understood _enough_ and I don't think we need to block this on eng feasibility further if that's "nice". I'll "unblock" this for product to finalize.

---

@sandysage I believe this is "needs-product" to finish up and make sure we have clear acceptance critera, designs, priority, etc.
